packages/astro/src/integrations/index.ts

```diff
@@ -49,7 +51,17 @@ export async function runHookConfigSetup({
 				injectRoute: (injectRoute) => {
 					updatedConfig._ctx.injectedRoutes.push(injectRoute);
 				},
-			});
+			}
+			// Semi-private `addPageExtension` hook
+			Object.defineProperty(hooks, 'addPageExtension', {
+				value: (...input: (string|string[])[]) => {
+					const exts = (input.flat(Infinity) as string[]).map(ext => `.${ext.replace(/^\./, '')}`);
+					updatedConfig._ctx.pageExtensions.push(...exts);
+				},
+				writable: false,
+				enumerable: false
+			})
```

Comment:
Not that private, but it works! It is intentionally not enumerable or exposed in the types, only there if you're looking for it.